internal/internal.go

```diff
@@ -506,8 +508,32 @@ func getRevision(ctx context.Context, store storage.Store, txn storage.Transacti
 	return revision, nil
 }
 
-func getParsedPath(req *ext_authz.CheckRequest) []interface{} {
+func getParsedQuery(req *ext_authz.CheckRequest) map[string]interface{} {
 	path := req.GetAttributes().GetRequest().GetHttp().GetPath()
+
+	parsedQuery := make(map[string]interface{})
```

Comment:
Let's use `net/url` to parse the URL here instead of rolling it from scratch. A side of effect of updating to use `net/url` is that the URL parameter values will be arrays of strings instead of just strings. This can't be avoided though (e.g., ?x=1&x=2 is valid.) We should also be able to parse the string once and construct the parsed path and query params at the same time.